### PR TITLE
Add HeadersToDownstreamOnAllow to ext_authz

### PR DESCRIPTION
**Pull request by hanxiaop**
First step to implement https://github.com/istio/istio/issues/36950

### CODE REVIEW

File: mesh/v1alpha1/config.proto [hunk 1]
```diff
@@ -619,8 +619,20 @@ message MeshConfig {
       // - Suffix match: "*abc" will match on value "abc" and "xabc".
       repeated string headers_to_downstream_on_deny = 8;
 
+      // List of headers from the authorization service that should be forwarded to downstream when the authorization
+      // check result is allowed (HTTP code 200).
+      // If not specified, the original request will not be modified and forwarded to backend as-is.
```
Comment (yangminzhu) [inline mesh/v1alpha1/config.proto:624]: `If not specified, the original request will not be modified and forwarded to backend as-is`

This is field is about the response and has nothing to do with the request to backend right?

Comment (hanxiaop) [reply mesh/v1alpha1/config.proto]: Thanks! Fixed.

File: mesh/v1alpha1/config.proto [hunk 2]
```diff
@@ -619,8 +619,20 @@ message MeshConfig {
       // - Suffix match: "*abc" will match on value "abc" and "xabc".
       repeated string headers_to_downstream_on_deny = 8;
 
+      // List of headers from the authorization service that should be forwarded to downstream when the authorization
+      // check result is allowed (HTTP code 200).
+      // If not specified, the original request will not be modified and forwarded to downstream as-is.
```
Comment (yangminzhu) [inline mesh/v1alpha1/config.proto:624]: "If not specified, the original **response** will not be modified ..."

This field is adding extra headers to the response to the downstream, please update the comment.

**Review by yangminzhu — APPROVED** [review]
Looks good, just a small comment.

File: mesh/v1alpha1/config.proto (hunk 2 above)
Comment (hanxiaop) [reply mesh/v1alpha1/config.proto]: Fixed. Thanks!

**Comment by hanxiaop** [comment]
@ericvn @howardjohn PTAL.

**Pull request merged**